tests/MongoDB.Driver.Tests/PinnedShardRouterTests.cs

```diff
@@ -176,7 +176,9 @@ private DisposableMongoClient CreateDisposableClient(EventCapturer eventCapturer
                 }, 
                 useMultipleShardRouters);
             var timeOut = TimeSpan.FromSeconds(60);
-            SpinWait.SpinUntil(() => client.Cluster.Description.Type != ClusterType.Unknown, timeOut).Should().BeTrue();
+            SpinWait.SpinUntil(
+                () => client.Cluster.Description.Servers.All(s => s.State == ServerState.Connected), timeOut)
```

Comment:
I won't object to this formatting, but...

we usually either have all arguments on one line or otherwise one argment per line.

Reply:
Refactored to use a local function :)